Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+
+"""
+Deephaven's learn module enables developers and data scientists to use familiar
+"""
+
+import jpy
+import wrapt
+
+_Input_ = None
+_Output_ = None
+_Computer_ = None
+_Scatterer_ = None
+
+
+def _defineSymbols():
+    """
+    Defines appropriate java symbol, which requires that the jvm has been initialized through the :class:`jpy` module,
+    for use throughout the module AT RUNTIME. This is versus static definition upon first import, which would lead to an
+    exception if the jvm wasn't initialized BEFORE importing the module.
+
+    Raises:
+        SystemError : if no JVM has been initialized.
+    """
+
+    if not jpy.has_jvm():
+        raise SystemError("No java functionality can be used until the JVM has been initialized through the jpy module")
+
+    global _Input_, _Output_, _Computer_, _Scatterer_
+    if _Input_ is None:
+        _Input_ = jpy.get_type("io.deephaven.integrations.learn.Input")
+        _Output_ = jpy.get_type("io.deephaven.integrations.learn.Output")
+        _Computer_ = jpy.get_type("io.deephaven.integrations.learn.Computer")
+        _Scatterer_ = jpy.get_type("io.deephaven.integrations.learn.Scatterer")
+
+
+# every module method that invokes Java classes should be decorated with @_passThrough
+@wrapt.decorator
+def _passThrough(wrapped, instance, args, kwargs):
+    """
+    For decoration of module methods, to define necessary symbols at runtime
+
+    Args:
+        wrapped  : the method to be decorated
+        instance : the object to which the wrapped function was bound when it was called
+        args     : the argument list for `wrapped`
+        kwargs   : the keyword argument dictionary for `wrapped`
+
+    Returns:
+        the decorated version of the method.
+    """
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+
+@_passThrough
+class Input:
+    """
+    Input specifies how to gather data from a Deephaven table into a Python object.
+    """
+    def __init__(self, colNames, gatherFunc):
+        """
+        Args:
+            colNames   : column name or list of column names from which to gather input.
+            gatherFunc : python function that determines how input gets transformed into a python object.
+        """
+        self.input = _Input_(colNames, gatherFunc)
+
+    def __str__(self):
+        return "Columns: " + str([self.input.getColNames()[i] for i in range(len(self.input.getColNames()))])
```

Comment:
I would implement the string in the java class as a `toString()` method.  Then call the method here.  Then either language can print out the details.